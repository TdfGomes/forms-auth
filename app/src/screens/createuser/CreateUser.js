import React, { useState } from 'react'
import { Button, Heading, TextInputField, Pane } from 'evergreen-ui'
import Container from '../../components/Container'

const initialState = {
  firstname: '',
  surname: '',
  email: '',
  password: ''
}

const CreateUser = () => {
  const [state, setState] = useState(initialState)
  const handleOnChange = e => {
    const {
      target: { name, value }
    } = e
    setState((prevState) => ({
      ...prevState,
      [name]: value
    }))
  }
  return (
    <Container background="tint1" minHeight="100vh" flexDirection="column">
      <Heading size={600} marginBottom={35}>
        Create an User
      </Heading>
      <Pane display="flex" justifyContent="space-between" width={500}>
        <Pane width="48%" marginBottom={25}>
          <TextInputField
            label="Firstname"
            name="firstname"
            placeholder="Firstname"
            onChange={handleOnChange}
            value={state.firstname}
          />

          <TextInputField
            label="Email Address"
            type="email"
            name="email"
            placeholder="Email address"
            onChange={handleOnChange}
            value={state.email}
          />
        </Pane>
        <Pane width="48%" marginBottom={25}>
          <TextInputField
            label="Surname"
            name="surname"
            placeholder="Surname"
            onChange={handleOnChange}
            value={state.surname}
          />
          <TextInputField
            label="Password"
            type="password"
            name="password"
            placeholder="Password"
            onChange={handleOnChange}
            value={state.password}
          />
        </Pane>
      </Pane>
      <Pane width={500} marginBottom={25} textAlign="right">
        <Button type="submit" onClick={() => alert(JSON.stringify(state))}>Submit</Button>
      </Pane>
    </Container>
  )
}

export default CreateUser
