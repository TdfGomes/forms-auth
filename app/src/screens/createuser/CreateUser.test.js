/* eslint-disable no-undef */
import React from 'react'
import { render } from 'react-testing-library'
import 'jest-dom/extend-expect'
import CreateUser from './CreateUser'
import { MockedProvider } from 'react-apollo/test-utils'

describe('<CreateUser/>', () => {
  it('Should render', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={[]}>
        <CreateUser />
      </MockedProvider>
    )
    expect(getByTestId('container')).toBeInTheDocument()
  })

  it('To Match Snapshot', () => {
    const { asFragment } = render(
      <MockedProvider mocks={[]}>
        <CreateUser />
      </MockedProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

