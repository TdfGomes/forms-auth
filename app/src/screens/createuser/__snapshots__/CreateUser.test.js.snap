// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateUser/> To Match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="ðŸ“¦bg_1vd85pq ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦min-h_100vh ðŸ“¦flx-drct_column ðŸ“¦box-szg_border-box css-nil"
    data-testid="container"
  >
    <h2
      class="ðŸ“¦mt_0px ðŸ“¦mb_35px ðŸ“¦fnt-sze_20px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_24px ðŸ“¦ltr-spc_-0-07px ðŸ“¦fnt-fam_wm9v6k ðŸ“¦color_234361 ðŸ“¦box-szg_border-box"
    >
      Create an User
    </h2>
    <form>
      <div
        class="ðŸ“¦dspl_flex ðŸ“¦just-cnt_space-between ðŸ“¦w_500px ðŸ“¦box-szg_border-box css-nil"
      >
        <div
          class="ðŸ“¦w_48prcnt ðŸ“¦mb_25px ðŸ“¦box-szg_border-box css-nil"
        >
          <div
            class="ðŸ“¦mb_24px ðŸ“¦box-szg_border-box"
          >
            <label
              class="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦dspl_block ðŸ“¦mb_4px ðŸ“¦box-szg_border-box"
              for="TextInputField-8"
            >
              Firstname 
            </label>
            <input
              aria-invalid="false"
              class="css-5ljhhe ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦w_100prcnt ðŸ“¦h_32px ðŸ“¦pl_10px ðŸ“¦pr_10px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦box-szg_border-box"
              id="TextInputField-8"
              name="firstname"
              placeholder="Firstname"
              spellcheck="true"
              type="text"
              value=""
            />
          </div>
          <div
            class="ðŸ“¦mb_24px ðŸ“¦box-szg_border-box"
          >
            <label
              class="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦dspl_block ðŸ“¦mb_4px ðŸ“¦box-szg_border-box"
              for="TextInputField-9"
            >
              Email Address 
            </label>
            <input
              aria-invalid="false"
              class="css-5ljhhe ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦w_100prcnt ðŸ“¦h_32px ðŸ“¦pl_10px ðŸ“¦pr_10px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦box-szg_border-box"
              id="TextInputField-9"
              name="email"
              placeholder="Email address"
              spellcheck="true"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="ðŸ“¦w_48prcnt ðŸ“¦mb_25px ðŸ“¦box-szg_border-box css-nil"
        >
          <div
            class="ðŸ“¦mb_24px ðŸ“¦box-szg_border-box"
          >
            <label
              class="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦dspl_block ðŸ“¦mb_4px ðŸ“¦box-szg_border-box"
              for="TextInputField-10"
            >
              Surname 
            </label>
            <input
              aria-invalid="false"
              class="css-5ljhhe ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦w_100prcnt ðŸ“¦h_32px ðŸ“¦pl_10px ðŸ“¦pr_10px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦box-szg_border-box"
              id="TextInputField-10"
              name="surname"
              placeholder="Surname"
              spellcheck="true"
              type="text"
              value=""
            />
          </div>
          <div
            class="ðŸ“¦mb_24px ðŸ“¦box-szg_border-box"
          >
            <label
              class="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦dspl_block ðŸ“¦mb_4px ðŸ“¦box-szg_border-box"
              for="TextInputField-11"
            >
              Password 
            </label>
            <input
              aria-invalid="false"
              class="css-5ljhhe ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦w_100prcnt ðŸ“¦h_32px ðŸ“¦pl_10px ðŸ“¦pr_10px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦box-szg_border-box"
              id="TextInputField-11"
              name="password"
              placeholder="Password"
              spellcheck="true"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="ðŸ“¦w_500px ðŸ“¦mb_25px ðŸ“¦txt-algn_right ðŸ“¦box-szg_border-box css-nil"
      >
        <button
          class="css-69cngj ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_16px ðŸ“¦pl_16px ðŸ“¦ml_0px ðŸ“¦mr_0px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
